pixeltable/catalog/table.py

```diff
@@ -276,53 +266,47 @@ def count(self) -> int:
         """Return the number of rows in this table."""
         return self._df().count()
 
-    @property
     def columns(self) -> list[str]:
         """Return the names of the columns in this table."""
         cols = self._tbl_version_path.columns()
         return [c.name for c in cols]
 
-    @property
     def _schema(self) -> dict[str, ts.ColumnType]:
         """Return the schema (column names and column types) of this table."""
         return {c.name: c.col_type for c in self._tbl_version_path.columns()}
 
-    @property
     def base_table(self) -> Optional['Table']:
-        with env.Env.get().begin_xact():
-            return self._base_table
+        from pixeltable.catalog import Catalog
+
+        with Catalog.get().begin_xact(for_write=False):
+            return self._base_table()
 
-    @property
     @abc.abstractmethod
     def _base_table(self) -> Optional['Table']:
-        """The base's Table instance"""
+        """The base's Table instance. Requires a transaction context"""
 
-    @property
     def _base_tables(self) -> list['Table']:
-        """The ancestor list of bases of this table, starting with its immediate base."""
-        bases = []
-        base = self._base_table
+        """The ancestor list of bases of this table, starting with its immediate base. Requires a transaction context"""
+        bases: list[Table] = []
+        base = self._base_table()
         while base is not None:
             bases.append(base)
-            base = base._base_table
+            base = base._base_table()
         return bases
 
     @property
     @abc.abstractmethod
     def _effective_base_versions(self) -> list[Optional[int]]:
         """The effective versions of the ancestor bases, starting with its immediate base."""
 
-    @property
     def _comment(self) -> str:
-        return self._tbl_version.get().comment
+        return self._tbl_version_path.comment()
 
-    @property
     def _num_retained_versions(self) -> int:
```

Comment:
Per discussion last week, a clearer name for these would be `_get_comment()`, `_get_num_retained_versions()`, etc.